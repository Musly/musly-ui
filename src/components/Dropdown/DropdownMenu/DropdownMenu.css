.dropdown__menu {
  position: absolute;
  left: 0;
  width: 100%;
  visibility: hidden;
  opacity: 0;
  transition-property: opacity;
  transition-duration: 200ms;

  @apply p-1 ease-sharp;

  &--top {
    top: 100%;
  }
}

.dropdown:hover .dropdown__menu {
  visibility: visible;
  opacity: 1;
}

.dropdown__menu__inner {
  @apply rounded-md bg-white py-1 shadow-lg border border-gray-100;
}

.dropdown__menu__inner__list {
  @apply my-1;
}
