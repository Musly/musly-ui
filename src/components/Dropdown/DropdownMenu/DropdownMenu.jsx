import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './DropdownMenu.css';

export function DropdownMenu ({ position, children }) {
  const menuClasses = classNames({
    dropdown__menu: true,
    'dropdown__menu--top': !position,
  }, position);

  return (
    <div className={menuClasses} data-test-id="dropdown-menu">
      <nav className="dropdown__menu__inner">
        <ul className="dropdown__menu__inner__list">
          {children}
        </ul>
      </nav>
    </div>
  );
}

DropdownMenu.propTypes = {
  children: PropTypes.node.isRequired,
  position: PropTypes.string,
};

DropdownMenu.defaultProps = {
  position: null,
};
