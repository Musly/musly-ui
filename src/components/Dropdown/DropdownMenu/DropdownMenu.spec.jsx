import React from 'react';
import { shallow } from 'enzyme';
import { DropdownMenu } from './DropdownMenu';

describe('<DropdownMenu />', () => {
  it('should render with default positioning', () => {
    const result = shallow(<DropdownMenu><div>Test</div></DropdownMenu>);
    expect(result.find('[data-test-id="dropdown-menu"]').hasClass('dropdown__menu--top')).toBe(true);
    expect(result).toMatchSnapshot();
  });

  it('should render with custom positioning', () => {
    const result = shallow(<DropdownMenu position="bottom-full"><div>Test</div></DropdownMenu>);
    expect(result.find('[data-test-id="dropdown-menu"]').hasClass('dropdown__menu--top')).toBe(false);
    expect(result.find('[data-test-id="dropdown-menu"]').hasClass('bottom-full')).toBe(true);
    expect(result).toMatchSnapshot();
  });
});
