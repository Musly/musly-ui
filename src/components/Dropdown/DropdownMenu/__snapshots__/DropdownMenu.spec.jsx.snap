// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownMenu /> should render with custom positioning 1`] = `
<div
  className="dropdown__menu bottom-full"
  data-test-id="dropdown-menu"
>
  <nav
    className="dropdown__menu__inner"
  >
    <ul
      className="dropdown__menu__inner__list"
    >
      <div>
        Test
      </div>
    </ul>
  </nav>
</div>
`;

exports[`<DropdownMenu /> should render with default positioning 1`] = `
<div
  className="dropdown__menu dropdown__menu--top"
  data-test-id="dropdown-menu"
>
  <nav
    className="dropdown__menu__inner"
  >
    <ul
      className="dropdown__menu__inner__list"
    >
      <div>
        Test
      </div>
    </ul>
  </nav>
</div>
`;
